### PR TITLE
EN-8219: ESDT API endpoints

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

**Review by miiu96 — CHANGES_REQUESTED** [review]

File: api/address/routes.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ func Routes(router *gin.RouterGroup) {
 	router.GET("/:address/shard", GetShard)
 	router.GET("/:address/transactions", GetTransactions)
 	router.GET("/:address/key/:key", GetValueForKey)
+	router.GET("/:address/esdt", GetESDTTokens)
+	router.GET("/:address/esdt/:tokenName", GetESDTTokenData)
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/address/routes.go:23]: done

File: api/address/routes.go [hunk 2]
```diff
@@ -188,3 +190,85 @@ func GetShard(c *gin.Context) {
...
+			c,
+			http.StatusBadRequest,
+			nil,
+			fmt.Sprintf("%v: %v", errors.ErrGetESDTTokenData, errors.ErrEmptyAddress),
+			data.ReturnCodeRequestError,
+		)
+		return
+	}
+
+	tokenName := c.Param("tokenName")
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/address/routes.go:214]: done

File: api/errors/errors.go [hunk 3]
```diff
@@ -14,12 +14,21 @@ var ErrGetValueForKey = errors.New("get value for key error")
...
+var ErrGetAllESDTTokens = errors.New("cannot get all ESDT tokens")
+
 // ErrEmptyAddress signals that an empty address was provided
 var ErrEmptyAddress = errors.New("address is empty")
 
 // ErrEmptyKey signals that an empty key was provided
 var ErrEmptyKey = errors.New("key is empty")
 
+// ErrEmptyTokenName signals that an token name was provided
+var ErrEmptyTokenName = errors.New("token name is empty")
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/errors/errors.go:30]: done

File: api/errors/errors.go [hunk 4]
```diff
@@ -14,12 +14,21 @@ var ErrGetValueForKey = errors.New("get value for key error")
...
+var ErrGetAllESDTTokens = errors.New("cannot get all ESDT tokens")
+
 // ErrEmptyAddress signals that an empty address was provided
 var ErrEmptyAddress = errors.New("address is empty")
 
 // ErrEmptyKey signals that an empty key was provided
 var ErrEmptyKey = errors.New("key is empty")
 
+// ErrEmptyTokenIdentifier signals that an empty token identifier was provided
+var ErrEmptyTokenIdentifier = errors.New("token name is empty")
```
Comment (miiu96) [inline api/errors/errors.go:30]: please change also the message of error

Comment (bogdan-rosianu) [reply api/errors/errors.go]: done

**Review by miiu96 — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**